helper.go

```diff
@@ -616,6 +620,10 @@ func parseLength(bytes []byte) (int, int, error) {
 				return 0, 0, ErrInvalidPacketLength
 			}
 			length += int(bytes[2+i])
+			if length < 0 {
+				// Invalid length due to overflow, return an error
```

Comment:
An integer overflow here would result in length being a negative value. In those cases the returned value of `length` could be less than the value of `cursor` and result in a `panic` in the calling code when using these values to slice data.

```
panic: runtime error: slice bounds out of range [109:108]

goroutine 1 [running]:
github.com/gosnmp/gosnmp.parseRawField({{0x0, 0x0}}, {0x140000c001a, 0xae, 0xae}, {0x104ea7f86, 0xb})
	/Users/username/git/gosnmp/helper.go:724 +0xa34
github.com/gosnmp/gosnmp.(*GoSNMP).unmarshalResponse(0x14000054de0, {0x140000c000e, 0xba, 0xba}, 0x140000c2000)
	/Users/username/git/gosnmp/marshal.go:1167 +0x648
github.com/gosnmp/gosnmp.(*GoSNMP).unmarshalPayload(0x14000054de0, {0x140000c0000, 0xc8, 0xc8}, 0xe, 0x140000c2000)
	/Users/username/git/gosnmp/marshal.go:1078 +0x164
github.com/gosnmp/gosnmp.(*GoSNMP).SnmpDecodePacket(0x14000054de0, {0x140000c0000, 0xc8, 0xc8})
	/Users/username/git/gosnmp/gosnmp.go:554 +0x174
main.main()
	/Users/username/git/gosnmp/examples/example-r0/main.go:12 +0x84
exit status 2
```

Reply:
Note that this was also brought up in PR #403.  I just added some commentary there. Adding this guard on `parseLength` probably avoids panics in various callers of `parseLength`.

As a note, this is tested by `panicUnmarshalParseRawFieldTimeTicks`.